### PR TITLE
Add Teleport Terminal tshd skeleton

### PR DESCRIPTION
**Pull request by codingllama**

### CODE REVIEW

File: lib/terminal/server.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+// Start creates and starts a Teleport Terminal service.
+// If ReadFromInput is present, then server a JSON-encoded version of opts is
+// read from ConfigInput, overriding any settings present.
+// If ConfigOutput is present, then a JSON-marshaled RuntimeOpts is sent to it
+// before server start.
+func Start(opts ServerOpts) (*Server, error) {
+	if opts.ReadFromInput {
+		if err := json.NewDecoder(opts.ConfigInput).Decode(&opts); err != nil {
```
Comment (zmb3) [inline lib/terminal/server.go:96]: You could check if `opts` is the zero value, and emit an error or at least log a warning that opts is being overwritten with the JSON input.

File: lib/terminal/server.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+		}
+	}
+
+	tlsOpts, err := parseTLSOpts(opts)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var network, addr string
+	switch tmp := strings.Split(opts.Addr, "://"); len(tmp) {
```
Comment (zmb3) [inline lib/terminal/server.go:107]: Can you use `utils.ParseAddr` here?

File: lib/terminal/server.go (hunk 1 above)
Comment (codingllama) [reply lib/terminal/server.go:96]: I think that is a valid scenario. It's up to Alexey, but I kind expect the usual invocation to be simply `tshd -stdin` with all the rest coming via stdin - seems simpler to do programmatically than passing flags.

File: lib/terminal/server.go (hunk 2 above)
Comment (codingllama) [reply lib/terminal/server.go:107]: Sounds really convenient, but ParseAddr returns empty Addr and Path for `unix://terminal.sock`, which seems broken to me: (`ParseAddr("unix://terminal.sock") = &utils.NetAddr{Addr:"", AddrNetwork:"unix", Path:""}`).

Not sure I want to get into that rabbit hole now.

**Pull request merged**